xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/api/Document.java

```diff
@@ -1229,6 +1263,21 @@ public Vector<Object> getObjects(String className)
         return getXObjects(objects);
     }
 
+    /**
+     * Retrieves and returns all objects corresponding to the class reference corresponding to the resolution of the
+     * given entity reference, or an empty list if there are none.
```

Comment:
I would specify in the javadoc that the list might contain null...

Reply:
It actually cannot contain `null` elements as they are filtered out by `getXObjects(objects)`. It's the same as for the method taking a `String`. If we want to change this, I think we should give the new method a new name in order to avoid that is accidentally called (some scripts might already pass an entity reference that is currently implicitly converted to string).

Reply:
Ah no it's good then, I missed the filter and I was being extra cautious...

Reply:
I guess in the future we should deprecate the API you replaced that are taking a String instead of a reference. But maybe not now...

Reply:
> But maybe not now...

Yeah, as usually with this kind of API, there is a lot of work before we can but the `@Dreprecated` annotation (if we don't want to explode the log). Now what we do sometimes is put at least the javadoc `@deprecated` (which does not have any log impact).